generate.py

```diff
@@ -48,6 +49,23 @@ def get_flags(configuration_name, _):
         return get_flags
     return get_factory
 
+def msvs_flags_factories(msvs):
+    def get_factory(category):
+        def get_flags(configuration_name, _):
+            if configuration_name is None:
+                return []
+            flags = []
+            if category == 'c':
+                flags += msvs.GetCflagsC(configuration_name)
+            elif category == 'cc':
+                flags += msvs.GetCflagsCC(configuration_name)
+            else:
+                raise RuntimeError('Unknown category: ' + category)
+            flags += msvs.GetCflags(configuration_name)
+            return flags
+        return get_flags
+    return get_factory
```

Comment:
This code looks the same as `xcode_flags_factories`. Maybe we can just rename it to `emulation_flags_factories` and pass in a `settings` parameter instead?

Reply:
I think, yes. Will make the edit.